DFC.ServiceTaxonomy.Taxonomies/Helper/TaxonomyHelper.cs

```diff
@@ -3,16 +3,24 @@
 using System.Collections.Generic;
 using System.Linq;
 using DFC.ServiceTaxonomy.Taxonomies.Models;
+using Microsoft.CSharp.RuntimeBinder;
 
 namespace DFC.ServiceTaxonomy.Taxonomies.Helper
 {
     public class TaxonomyHelper : ITaxonomyHelper
     {
         public List<dynamic>? GetTerms(dynamic contentItem)
         {
-            return contentItem.Content?[nameof(TaxonomyPart)]?[nameof(TaxonomyPart.Terms)]?.ToObject<List<dynamic>?>() ??
-                   contentItem.Content?[nameof(TaxonomyPart.Terms)]?.ToObject<List<dynamic>?>() ??
-                   contentItem[nameof(TaxonomyPart.Terms)]?.ToObject<List<dynamic>?>();
+            try
+            {
+                return contentItem.Content?[nameof(TaxonomyPart)]?[nameof(TaxonomyPart.Terms)]?.ToObject<List<dynamic>?>() ??
+                       contentItem.Content?[nameof(TaxonomyPart.Terms)]?.ToObject<List<dynamic>?>() ??
+                       contentItem[nameof(TaxonomyPart.Terms)]?.ToObject<List<dynamic>?>();
+            }
+            catch (RuntimeBinderException)
```

Comment:
Not the greatest way of handling it. Would be better to check context without an exception being thrown (using `ContainsKey`?).

Reply:
Not possible on a dynamic as far as I'm aware! Tried casting to a JObject but that also blew up.